ldms/src/core/ldms.h

```diff
@@ -1206,6 +1218,30 @@ uint32_t ldms_set_perm_get(ldms_set_t s);
  */
 int ldms_set_perm_set(ldms_set_t s, mode_t perm);
 
+#define DEFAULT_AUTHZ_SET_UID 0x4
+#define DEFAULT_AUTHZ_SET_GID 0x2
+#define DEFAULT_AUTHZ_SET_PERM 0x1
+#define DEFAULT_AUTHZ_SET_ALL (SET_DEFAULT_AUTHZ_UID|SET_DEFAULT_AUTHZ_GID|SET_DEFAULT_AUTHZ_PERM)
+#define DEFAULT_AUTHZ_READONLY 0
+/**
+ * \brief Atomically set or get one or more default authorization values for LDMS sets.
+ *
+ * The uid, gid, and perm options can each individually be inputs or outputs. To make
+ * uid, gid, or perm as inputs set the bits DEFAULT_AUTHZ_SET_UID,
+ * DEFAULT_AUTHZ_SET_GID, or SET_DEFAULT_SET_AUTHZ_PERM bits respectively in the
+ * set_flags option.  DEFAULT_AUTHZ_SET_ALL is provided as a convience to set all three
+ * bits.
+ *
+ * Any option for which the associated set_flags is not set will be an _output_,
+ * atomically reporting the current value.
+ *
+ * \param uid UID default to set if SET_DEFAULT_AUTHZ_UID bit is set in set_flags
+ * \param gid GID default to set if SET_DEFAULT_AUTHZ_GID bit is set in set_flags
+ * \param perm Permissions default to set if SET_DEFAULT_AUTHZ_PERM bit is set in set_flags
+ * \param set_flags
+ */
+void ldms_set_default_authz(uid_t *uid, gid_t *gid, mode_t *perm, int set_flags);
```

Comment:
Wouldn't two separate APIs for getting and setting be easier to use? Callers do not have to pass any flags. For example,
```
int ldms_set_default_authz_set(uid_t uid, gid_t gid, mode_t perm)
void ldms_set_default_authz_get(uid_t *uid, gid_t *gid, mode_t *perm)
```
For the set API, callers would pass -1 to the argument(s) that they want to ignore.
For the get API, callers would pass NULL to the argument(s) that they don't want the value.

Reply:
I'm not as comfortable using -1 for uid/gid. Can you show me that those are not valid IDs? That is the main reason for the flags.

A separate set/get would not allow for the ability to atomically know how the defaults are set after setting a subset of the values. The "set" function would need to remain largely as it is with parameters that can be both inputs and outputs.

I suppose I could add a separate _get() wrapper for convenience if people really think it is required.